rts/System/Sound/OpenAL/Sound.cpp

```diff
@@ -493,40 +493,58 @@ void CSound::OpenLoopbackDevice(const std::string& deviceName)
 		return;
 	}
 
+	LOG("[Sound::%s] SDL audio device(s): ", __func__);
+	for (int i = 0, n = SDL_GetNumAudioDevices(0); i < n; ++i) {
+		LOG("[Sound::%s]  * \"%d\" \"%s\"", __func__, i, SDL_GetAudioDeviceName(i, 0));
+	}
+
 	SDL_AudioSpec desiredSpec;
 	SDL_AudioSpec obtainedSpec;
 
-	desiredSpec.channels = 2;
 	desiredSpec.format = AUDIO_S16SYS;
 	desiredSpec.freq = 44100;
 	desiredSpec.padding = 0;
 	desiredSpec.samples = 4096;
 	desiredSpec.callback = RenderSDLSamples;
 	desiredSpec.userdata = this;
+	
+	/* SDL bug: can return devices with >2 channels (3D surround), even if we ask for just 2.
+ 	* (if tested) This causes the 2 "primary" channels to be moved in the 3D space compared to their "normal" state.
+ 	* (if not tested) We are afraid that this risks the 2 "primary" channels to be moved in the 3D space compared to their "normal" state, but this remains to be tested.
+ 	* For this reason, the engine rejects such devices down the road. Don't let it get that far and retry instead.
+ 	*
+ 	* Note that proper support for 3D surround sounds sounds hard, for example as of 2023 counter-strike has very weak support
+ 	* for it, apparently noticeably worse than just stereo according to players, despite being in a more relevant genre. */
+	for(int channelsDesired = 2; channelsDesired > 0; channelsDesired--) {
+		desiredSpec.channels = channelsDesired;
+
+		sdlDeviceID = 0;
+		selectedDeviceName = "";
+
+		if (!deviceName.empty()) {
```

Comment:
Respectively this can go outside of loop too, as `deviceName.empty()` value doesn't change.

Reply:
deviceName is set by user in springsettings.cfg. On default it's empty.
When it's set, we preferably try to open that desired device but fall back to default device on failure. 
This fallback is handled by recognizing sdlDeviceID is 0.

1. selectedDeviceName = "";
This could be moved outside the loop. It's just overwritten with any opened SDL device and not needed at all, if we don't want that opened sdl devie for any reason.

2. 
I don't see how any of the other 2 lines could be excluded without changing the logic in the for-loop.
a. if we obtain too many channels, we still set sdlDeviceID. So it needs to be 0 for next try.
b. On next try we need to check if deviceName is empty again, because we still try to follow preference of user.

Reply:
Yeah makes sense.